miso-web/src/main/webapp/scripts/array_ajax.js

```diff
@@ -0,0 +1,389 @@
+        if (position) {
+          $('#selectedPosition').text(position);
+        } else {
+          $('#selectedPosition').empty();
+          $('#search, #searchField, #resultSelect, #updateSelected').prop('disabled', true).addClass('disabled');
+        }
+        $('#selectedName').empty();
+        $('#selectedAlias').empty();
+        $('#selectedBarcode').empty();
+        $('#updateSelected, #removeSelected').prop('disabled', true).addClass('disabled');
+      }
+      $('#warningMessages').html('');
+      $('#searchField').val('');
+      $('#searchField').select().focus();
+    };
+
+    self.getRowLabel = function(row) {
+      return row >= 10 ? row : '0' + row;
+    };
+
+    self.getColLabel = function(col) {
+      return positionStringifier.getRowLabel(col);
+    };
+
+    self.getPositionString = function(row, col) {
+      return 'R' + positionStringifier.getRowLabel(row) + 'C' + positionStringifier.getColLabel(col);
+    };
+
+    return self;
+  };
+
+  SampleArray.setArrayJson = function(json) {
+    arrayJson = json;
+    if (!visual) {
+      visual = new SampleArray.Visual();
+    }
+    updatePage();
+  };
+
+  SampleArray.validateAndSave = function() {
+    Validate.cleanFields('#array-form');
+    Validate.clearErrors('#array-form');
+
+    // ArrayModel input field validation
+    $('#arrayModel').attr('class', 'form-control');
+    $('#arrayModel').attr('data-parsley-required', 'true');
+    $('#arrayModel').attr('data-parsley-errors-container', '#arrayModelError');
+
+    // Alias input field validation
+    $('#alias').attr('class', 'form-control');
+    $('#alias').attr('data-parsley-required', 'true');
+    $('#alias').attr('data-parsley-maxlength', '255');
+    $('#alias').attr('data-parsley-pattern', Utils.validation.sanitizeRegex);
+    $('#alias').attr('data-parsley-errors-container', '#aliasError');
+
+    // Serial Number input field validation
+    $('#serialNumber').attr('class', 'form-control');
+    $('#serialNumber').attr('data-parsley-required', 'true');
+    $('#serialNumber').attr('data-parsley-maxlength', '255');
+    $('#serialNumber').attr('data-parsley-pattern', Utils.validation.sanitizeRegex);
+    $('#serialNumber').attr('data-parsley-errors-container', '#serialNumberError');
+
+    // Description input field validation
+    $('#description').attr('class', 'form-control');
+    $('#description').attr('data-parsley-max-length', '255');
+    $('#description').attr('data-parsley-pattern', Utils.validation.sanitizeRegex);
+    $('#description').attr('data-parsley-errors-container', '#descriptionError');
+
+    $('#array-form').parsley();
+    $('#array-form').parsley().validate();
+
+    Validate.updateWarningOrSubmit('#array-form', null, function() {
+      save(!arrayJson || !arrayJson.id);
+    });
+  };
+
+  SampleArray.removeSelected = function() {
+    var selectedPosition = visual.selectedItems[0].position;
+    var selectedItem = getItemAtPosition(selectedPosition);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "selectedItem" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'selectedItem' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)